tests/cpp_unit_tests/test_y_bus.cpp

```diff
@@ -272,6 +274,181 @@ TEST_CASE("Test fill-in y bus") {
+        {2, 3}, // branch 2 from node 2 to 3
+        {3, 2}, // branch 3 from node 3 to 2
+        {0, 1}, // branch 4 from node 0 to 1
+        {2, -1} // branch 5 from node 2 to "not connected"
+    };
+    MathModelParam<true> param_sym_update;
+    param_sym_update.branch_param = {                        //   ff,    ft,   tf,   tt
+                                     {1.0i, 0.0, 0.0, 0.0},  // {1,  0}
+                                     {0.0, 1.0, 0.0, 0.0},   // {1,  2}
+                                     {0.0, 0.0, 0.0, 2.0i},  // {2,  3}
+                                     {0.0, 0.0, 2.0, 0.0},   // {3,  2}
+                                     {0.0, 0.0, 0.0, 0.0},   // {0,  1}
+                                     {1.0i, 0.0, 0.0, 0.0}}; // {2, -1}
+    param_sym_update.shunt_param = {1.0i, 0.0i};
+
+    const ComplexTensorVector<true> admittance_sym_2 = {
+        // 17.0 + 104.0i,                                                                        [v]
+        17.0 + 105.0i, // 0, 0 -> += {1, 0}tt + {0, 1}ff + shunt(0) = 0.0 + 0.0 + 1.0i
+        // 18.0 + 3.0i,                                                                          [v]
+        18.0 + 3.0i, // 0, 1 -> += {0, 1}ft + {1, 0}tf = 0.0 + 0.0
+        // 19.0 + 2.0i,                                                                          [v]
+        19.0 + 2.0i, // 1, 0 -> += {0, 1}tf + {1, 0}ft = 0.0 + 0.0
+        // 25.0 + 1.0i,                                                                          [v]
+        25.0 + 2.0i, // 1, 1 -> += {0, 1}tt + {1, 0}ff + {1,2}ff = 0.0 + 1.0i + 0.0
+        // 6.0,                                                                                  [v]
+        7.0, // 1, 2 -> += {1,2}ft = 1.0
+        // 7.0,                                                                                  [v]
+        7.0, // 2, 1 -> += {1,2}tf = 0.0
+        // 24.0 + 1009.0i,                                                                       [v]
+        24.0 + 1010.0i, // 2, 2 -> += {1,2}tt + {2,3}ff + {3, 2}tt + {2,-1}ff = 0.0 + 0.0 + 0.0 + 1.0i
+        // 15.0 + 10.0i,                                                                         [v]
+        17.0 + 10.0i, // 2, 3 -> += {2,3}ft + {3,2}tf = 0.0 + 2.0
+        // 14.0 + 11.0i,                                                                         [v]
+        14.0 + 11.0i, // 3, 2 -> += {2,3}tf + {3,2}ft = 0.0 + 0.0
+        // 13.0 + 212.0i                                                                         [v]
+        13.0 + 214.0i // 3, 3 -> += {2,3}tt + {3,2}ff + shunt(1) = 2.0i + 0.0 + 0.0
+    };
+
+    SUBCASE("Test whole scale update") {
+        YBus<true> ybus{topo_ptr, std::make_shared<MathModelParam<true> const>(param_sym)};
+        CHECK(ybus.admittance().size() == admittance_sym.size());
+        for (size_t i = 0; i < admittance_sym.size(); i++) {
+            CHECK(cabs(ybus.admittance()[i] - admittance_sym[i]) < numerical_tolerance);
```

Comment:
this numerical tolerance check will fail except for some cases if you subtract two very large values resulting in drift for very many iterations. Did you verify that the case you have here indeed causes drift when implementing it naively (i.e. by just adding the difference between the original and updated value)?

Reply:
I'll craft such a test case and see. See [above](https://github.com/PowerGridModel/power-grid-model/pull/444#discussion_r1433025454), also subject to change.